docs/end-user-topics/dois.rst

```diff
@@ -0,0 +1,91 @@
+DOIs
+====
+
+Introduction
+------------
+A Digial Object Identifier (DOI) is a permanent identifier that can be used in publications to identify the exact
+version of a workflow or tool (entry). DOIs are commonly referenced from academic papers.
+See the `DOI Foundation <https://www.doi.org>`__ for more details.
+
+A Dockstore entry that has a DOI will have a concept DOI, which is a DOI referencing the entire entry, and potentially many version DOIs -- one DOI per version. The concept DOI represents all versions and will resolve to the latest version.
```

Comment:
> I recently learned that concept DOI seems to be a zenodo concept. So far, this sentence seems to be ok even with harvested DOIs since GitHub's integration also uses zenodo, but it could be different someday if we work with other DOIs.

Yes, this is important to understand, so please permit me to reiterate:

A DOI itself, per the standard and infrastructure, is simply an identifier which is associated with some summary metadata which typically includes a link to a resource.  The DOI standard supports updating the DOI's metadata after it is created.

Zenodo has created the notion of a concept DOI and version DOI, and also archives the resource, and Zenodo-issued DOIs operate within this "semi-permanent archive" conceptual framework.

However, many [non-Zenodo] DOIs simply link directly to the original resource (paper PDF on the publishers site, for example), and there is no archival copy made.

Reply:
To be clear, I think concept DOIs (the concept of organizing and linking versioned DOIs) are important and worth bringing up in this document (as are those archives). 

I'm just bringing this up in the context of making the doc future-proof.

Reply:
Hmm, I did not know this. I will tailor the doc accordingly.